src/cpuinfo_x86.c

```diff
@@ -1135,6 +1158,9 @@ X86Info GetX86Info(void) {
   if (IsVendor(leaf_0, "GenuineIntel") || IsVendor(leaf_0, "AuthenticAMD")) {
```

Comment:
Here let's instantiate a struct like this:
```
struct OsSupport {
  bool have_sse_os_support;
  bool have_avx_os_support;
};
```

Then pass the struct by pointer to `ParseCpuId` so it can fill it, and pass the struct by value to `ParseExtraAMDCpuId` so it can read it. This way we don't redo the work (each call to cpuid is ~100 cycles)

Reply:
done